pages/docs/couple-your-code/couple-your-code-waveform.md

```diff
@@ -0,0 +1,167 @@
+
+The experimental API has to be activated in the configuration file via the `experimental` attribute. This allows us to define the order of the interpolant in the `read-data` tag of the corresponding `participant`. Here, `waveform-order="1"` corresponds to linear interpolation and `waveform-order="0"` to constant interpolation (this is just the same as the stable API).
+
+```xml
+<solver-interface experimental="true" ... >
+...
+    <participant name="FluidSolver">
+        <use-mesh name="FluidMesh" provide="yes"/>
+        <write-data name="Forces" mesh="MyMesh"/>
+        <read-data name="Displacements" mesh="FluidMesh" waveform-order="1"/>
+    </participant>
+...
+</solver-interface>
+```
+
+## Usage examples
+
+Let's use this experimental API in two examples from the step-by-step guide.
+
+### Use waveform iteration without subcycling
+
+We begin with an extended version of the example from ["Step 3 - Mesh and data access"](couple-your-code-timestep-sizes.html). Only few changes are necessary to sample the `Displacements` at the middle of the time window, which might be necessary for our specific application:
+
+```cpp
+turnOnSolver(); //e.g. setup and partition mesh 
+
+precice::SolverInterface precice("FluidSolver","precice-config.xml",rank,size); // constructor
+
+int dim = precice.getDimensions();
+int meshID = precice.getMeshID("FluidMesh");
+int vertexSize; // number of vertices at wet surface 
+// determine vertexSize
+double* coords = new double[vertexSize*dim]; // coords of coupling vertices 
+// determine coordinates
+int* vertexIDs = new int[vertexSize];
+precice.setMeshVertices(meshID, vertexSize, coords, vertexIDs); 
+delete[] coords;
+
+int displID = precice.getDataID("Displacements", meshID); 
+int forceID = precice.getDataID("Forces", meshID); 
+double* forces = new double[vertexSize*dim];
+double* displacements = new double[vertexSize*dim];
+
+double dt;         // solver timestep size
+double precice_dt; // maximum precice timestep size
+
+precice_dt = precice.initialize();
+while (not simulationDone()){ // time loop
+  dt = beginTimeStep(); // e.g. compute adaptive dt 
+  dt = min(precice_dt, dt);
+  // sampling in the middle of the time step
+  precice.readBlockVectorData(displID, vertexSize, vertexIDs, 0.5 * dt, displacements);
+  setDisplacements(displacements); // displacement at the middle of the time step
+  solveTimeStep(dt); // might be using midpoint rule for time-stepping
+  computeForces(forces);
+  precice.writeBlockVectorData(forceID, vertexSize, vertexIDs, forces);
+  precice_dt = precice.advance(dt);
+  endTimeStep(); // e.g. update variables, increment time
+}
+precice.finalize();
+delete[] vertexIDs, forces, displacements;
+turnOffSolver();
+```
```

Comment:
I guess you could leave out the obvious parts here and abbreviate with `[...]`.

Reply:
Done. 07fb39cf7a8fd05ab3cf9afbb854130f57e703cf